packages/framework/fluid-static/src/types.ts

```diff
@@ -47,31 +47,10 @@ export type LoadableObjectClass<T extends IFluidLoadable = IFluidLoadable> =
  * @typeParam T - The class of the `DataObject`.
  * @public
  */
-export type DataObjectClass<T extends IFluidLoadable> = {
+export type DataObjectClass<T extends IFluidLoadable = IFluidLoadable> = {
 	readonly factory: { IFluidDataStoreFactory: DataObjectClass<T>["factory"] };
-} & LoadableObjectCtor<T>;
-
-/**
- * A factory that produces a factory that can create a DDSes (`SharedObject`s).
- *
- * @typeParam T - The class of the `SharedObject`.
- * @public
- */
-export interface SharedObjectClass<T extends IFluidLoadable> {
-	/**
-	 * Gets the factory this factory is a wrapper for.
-	 */
-	readonly getFactory: () => IChannelFactory<T>;
-}
-
-/**
- * An object with a constructor that will return an {@link @fluidframework/core-interfaces#IFluidLoadable}.
- *
- * @typeParam T - The class of the loadable object.
- * @public
- */
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type LoadableObjectCtor<T extends IFluidLoadable> = new (...args: any[]) => T;
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+} & (new (...args: any[]) => T);
```

Comment:
Do we even need the constructor included here? If the factories are the only things that should be allowed to call new, and they know the explicit type of the thing they're newing, seems this should be hideable.

Reply:
DataObjectClass really needs a refactor in that direction, but I think fixing that is beyond the scope of this PR.

As the data object APIs needed to use this are all alpha, fixing this is lower priority, and I don't think its as simple as removing the constructor (since the "factory" member doesn't actually provider any way to create an instance of the object or a way to know what parameters would be needed). I'd like to leave that for another PR as its not clear to me exactly what should be done, and it impacts a lot of code this PR doesn't touch.

Note that it is possible to implement this API without exposing the concreate class type (you just have to type the class object as an instance of DataObjectClass<theInterface>, so its not as bad as it seems, but I still think we should change it at some point.